### PR TITLE
feat(batcher): Update SP1 and Risc0 version in batcher, examples, and docs

### PR DESCRIPTION
**Pull request by PatStiles**
# Update Batcher SP1 and Risc0 version

## Description

Complements #1324 
Updates the SP1 and Risc0 version in the:
- batcher
- examples
- docs
Renames the test files for SP1 and Risc0 by changing the respective suffix to point towards the correct version ("" indicates removal of the suffix):
"_new" -> ""
"" -> "-> old

NOTE: Removal of the old verifier ffi bindings will occur in a third pr.

## Type of change

- [x] New feature

### CODE REVIEW

File: docs/3_guides/4_generating_proofs.md [hunk 1]
```diff
@@ -303,6 +303,6 @@ To generate a proof of the execution of your code run the following:
   ```sh
   cargo run --release -- prove-risc0  <PROGRAM_DIRECTORY_PATH> .
   ```
-  Make sure to have [Risc0](https://dev.risczero.com/api/zkvm/quickstart#1-install-the-risc-zero-toolchain) installed with version `v1.0.1`
+  Make sure to have [Risc0](https://dev.risczero.com/api/zkvm/quickstart#1-install-the-risc-zero-toolchain) installed with version `v1.1.2`
```
Comment (entropidelic) [inline docs/3_guides/4_generating_proofs.md]: we should also mention about the `metal` installation for risc zero in MacOS

**Review by MauroToscano — CHANGES_REQUESTED** [review]
It's approved, but don't merge until operators upgrade

**Review by JuArce — APPROVED** [review]
Cases tested in stage:

Operator version: v0.10.2

1. Send old proofs with the old batcher: It is the previous version, and it works. The operator logs that it is using old version of the prover ✅
2. Send new proofs with the old batcher: The batcher rejects the proofs ✅
3. Send new proofs with the new batcher. The operator uses the new version of the provers ✅

**Review by MarcosNicolau — APPROVED** [review]
Worked as expected on Linux and macos!

**Comment by Oppen** [comment]
> It's approved, but don't merge until operators upgrade

Updates on this?

**Comment by uri-99** [comment]
Still waiting , aprox 24hs and we eject unupgraded operators.

**Review by MauroToscano — APPROVED** [review]

**Pull request merged**